[2024-11-05T22:59:20.324+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T22:59:19.041984+00:00 [queued]>
[2024-11-05T22:59:20.331+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T22:59:19.041984+00:00 [queued]>
[2024-11-05T22:59:20.332+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-11-05T22:59:20.343+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): transform> on 2024-11-05 22:59:19.041984+00:00
[2024-11-05T22:59:20.349+0000] {standard_task_runner.py:57} INFO - Started process 89 to run task
[2024-11-05T22:59:20.352+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_transform', 'transform', 'manual__2024-11-05T22:59:19.041984+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/spark_transform.py', '--cfg-path', '/tmp/tmp55d7n96s']
[2024-11-05T22:59:20.353+0000] {standard_task_runner.py:85} INFO - Job 3: Subtask transform
[2024-11-05T22:59:20.411+0000] {task_command.py:416} INFO - Running <TaskInstance: spark_transform.transform manual__2024-11-05T22:59:19.041984+00:00 [running]> on host 583aba4a4002
[2024-11-05T22:59:20.482+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_transform' AIRFLOW_CTX_TASK_ID='transform' AIRFLOW_CTX_EXECUTION_DATE='2024-11-05T22:59:19.041984+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-05T22:59:19.041984+00:00'
[2024-11-05T22:59:20.494+0000] {base.py:73} INFO - Using connection ID 'SPARK_CONNECTION' for task execution.
[2024-11-05T22:59:20.495+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /example.py
[2024-11-05T22:59:20.500+0000] {spark_submit.py:536} INFO - /home/***/.local/bin/spark-submit: line 27: /opt/spark/bin/spark-class: No such file or directory
[2024-11-05T22:59:20.512+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /example.py. Error code is: 127.
[2024-11-05T22:59:20.515+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spark_transform, task_id=transform, execution_date=20241105T225919, start_date=20241105T225920, end_date=20241105T225920
[2024-11-05T22:59:20.524+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 3 for task transform (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /example.py. Error code is: 127.; 89)
[2024-11-05T22:59:20.564+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-11-05T22:59:20.582+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
